docs/separate-source-sets.md

```diff
@@ -0,0 +1,95 @@
+# Benchmarking with Gradle: Creating Separate Source Sets
+
+This guide will walk you through the process of creating separate source sets for benchmarks in your Kotlin project with Gradle. It aims to help you organize, streamline, and isolate your benchmarking code effectively.
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+2. [What is a Source Set?](#what-is-a-source-set)
+3. [Why Have Separate Source Sets for Benchmarks?](#why-have-separate-source-sets-for-benchmarks)
+4. [Step-by-step Setup Guide](#setup-guide)
+   - [Step 1: Defining the Source Set](#step-1)
+   - [Step 2: Adding Dependencies](#step-2)
+   - [Step 3: Registering the Benchmark Source Set](#step-3)
+5. [Frequently Asked Questions](#frequently-asked-questions)
+6. [Troubleshooting](#troubleshooting)
+7. [Conclusion](#conclusion)
+
+## Introduction <a name="introduction"></a>
+
+This guide provides a step-by-step process for creating separate source sets dedicated to benchmarks in your Kotlin project using Gradle. By following this guide, you'll be able to maintain a clean and organized project structure, while ensuring efficient benchmarking for your code.
+
+## What is a Source Set? <a name="what-is-a-source-set"></a>
+
+Before we delve into the details, let's clarify what a source set is. In Gradle, a source set represents a group of source files that are compiled and executed together. By default, every Gradle project includes two source sets: `main` for your application code and `test` for your test code.
+
+A source set defines the location of your source code, the names of compiled classes, and their placement. It also handles additional assets such as resources and configuration files.
+
+## Why Have Separate Source Sets for Benchmarks? <a name="why-have-separate-source-sets-for-benchmarks"></a>
+
+Having separate source sets for benchmarks offers several advantages:
+
+1. **Organization**: It helps maintain a clean and organized project structure. Segregating benchmarks from the main code makes it easier to navigate and locate specific code segments.
+
+2. **Isolation**: Separating benchmarks ensures that the benchmarking code does not interfere with your main code or test code. This isolation guarantees accurate measurements without unintentional side effects.
+
+3. **Flexibility**: Creating a separate source set allows you to manage your benchmarking code independently. You can compile, test, and run benchmarks without impacting your main source code.
+
+## Step-by-step Setup Guide <a name="setup-guide"></a>
+
+Setting up a separate source set for benchmarks in your Kotlin project involves the following key steps:
```

Comment:
It's worth mentioning that the steps work for JVM-only projects.
There is an active effort to support separate multiplatform source sets for benchmarks: https://github.com/Kotlin/kotlinx-benchmark/pull/112
And steps for setting it up are a bit different.

Reply:
Noted, I have addressed this in the latest commit now that the pr has been merged.